lib/rofl_ofdpa_fm_driver.cpp

```diff
@@ -2412,8 +2412,9 @@ cofgroupmod rofl_ofdpa_fm_driver::enable_group_l3_unicast(
   uint32_t group_id = group_id_l3_unicast(id);
   cofgroupmod gm(ofp_version);
 
-  assert(0 == get_group_type(l2_interface) &&
-         "wrong l2 interface in enable_group_l3_unicast");
+  assert(0 == get_group_type(l2_interface) ||
+      11 == get_group_type(l2_interface) ||
+      12 == get_group_type(l2_interface));
```

Comment:
You are adding both l2 unfiltered (11) and trunk (12), but don't add trunk unfiltered (13). Is this intentional? Also adding l2 unfiltered does not match the prefix.

So either add trunk unfiltered as well or drop l2 unfiltered from the allowed list.

Also a short comments explaining the magic numbers might be nice.

```suggestion
  assert(0 == get_group_type(l2_interface) || /* l2 */
      12 == get_group_type(l2_interface)); /* trunk */
```
or

```suggestion
  assert(0 == get_group_type(l2_interface) || /* l2 */
      11 == get_group_type(l2_interface) || /* l2 unfiltered */
      12 == get_group_type(l2_interface) || /* trunk */
      13 == get_group_type(l2_interface)); /* trunk unfiltered */
```